graph_search.py

```diff
@@ -0,0 +1,115 @@
+class GraphSearch:
+  def dfs(self, target, graph, starting_vertex):
+    info = {
+      "target": target,
+      "found": False,
+      "visited": [],
+      "checks_whether_vertex_is_target": 0
+    }
+
+    def traverse(vertex):
+      info["visited"].append(vertex)
+      info["checks_whether_vertex_is_target"] += 1
+      if (vertex == target):
+        info["found"] = True
+        return info
+
+      for v in graph[vertex]:
+        if v not in info["visited"]:
+          if traverse(v)["found"] == True:
+            return info
+
+      return info
+
+    traverse(starting_vertex)
+    return info
+
+  def bfs(self, target, graph, starting_vertex):
+    from collections import deque
+
+    info = {
+      "target": target,
+      "found": False,
+      "visited": [],
```

Comment:
I've found it helpful to give a clear description of what "node is in visited" means.  For example, does it mean:

- I've added the node to the queue (and don't need to add it again)
- I've dequeued the node from the queue and checked it (so if I dequeue the same node again, I can ignore it)